src/session.rs

```diff
@@ -853,4 +860,16 @@ impl Session {
             m.clear_send_buffers();
         }
     }
+
+    fn configure_pacer_padding(
+        pacer: &mut PacerImpl,
+        last_estimate: Option<Bitrate>,
+        desired_bitrate: Bitrate,
+    ) {
+        let padding_rate = last_estimate
+            .map(|bwe| (bwe * PADDING_FACTOR).min(desired_bitrate))
+            .unwrap_or(Bitrate::ZERO);
+
+        pacer.set_padding_rate(padding_rate);
+    }
```

Comment:
This looks too complicated. Consider doing something like this:

```diff
diff --git a/src/session.rs b/src/session.rs
index 77dee7a..3865470 100644
--- a/src/session.rs
+++ b/src/session.rs
@@ -69,6 +69,7 @@ pub(crate) struct Session {
     twcc_tx_register: TwccSendRegister,
 
     bwe: Option<SendSideBandwithEstimator>,
+    bwe_desired_bitrate: Bitrate,
 
     enable_twcc_feedback: bool,
 
@@ -130,6 +131,7 @@ impl Session {
             twcc_rx_register: TwccRecvRegister::new(100),
             twcc_tx_register: TwccSendRegister::new(1000),
             bwe,
+            bwe_desired_bitrate: Bitrate::ZERO,
             enable_twcc_feedback: false,
             pacer,
             poll_packet_buf: vec![0; 2000],
@@ -512,6 +514,7 @@ impl Session {
                     let records = range.and_then(|range| self.twcc_tx_register.send_records(range));
 
                     if let Some(records) = records {
+                        self.configure_pacer_padding();
                         bwe.update(records, now);
                     }
                 }
@@ -826,14 +829,8 @@ impl Session {
     }
 
     pub fn set_bwe_desired_bitrate(&mut self, desired_bitrate: Bitrate) {
-        if let Some(bwe) = &mut self.bwe {
-            let padding_rate = bwe
-                .last_estimate()
-                .map(|bwe| (bwe * PADDING_FACTOR).min(desired_bitrate))
-                .unwrap_or(Bitrate::ZERO);
-
-            self.pacer.set_padding_rate(padding_rate);
-        }
+        self.bwe_desired_bitrate = desired_bitrate;
+        self.configure_pacer_padding();
     }
 
     pub fn line_count(&self) -> usize {
@@ -853,4 +850,18 @@ impl Session {
             m.clear_send_buffers();
         }
     }
+
+    fn configure_pacer_padding(&mut self) {
+        let Some(last_estimate) = self.bwe.as_ref().map(|b| b.last_estimate()) else {
+            return;
+        };
+
+        let padding_rate = if let Some(last) = last_estimate {
+            (last * PADDING_FACTOR).min(self.bwe_desired_bitrate)
+        } else {
+            self.bwe_desired_bitrate
+        };
+
+        self.pacer.set_padding_rate(padding_rate);
+    }
 }
```

Reply:
I did something like this initially but had problems with overlapping `&mut self` borrows. Does that actually compile?

Reply:
Does indeed compile

Reply:
Hmm, let me try again

Reply:
I still get a conflicting borrow between `records` and the call to `configure_pacer_padding` but fixed it slightly differently